packages/plugin-ext/src/main/browser/menus/plugin-menu-command-adapter.ts

```diff
@@ -230,6 +235,30 @@ export class PluginMenuCommandAdapter implements MenuCommandAdapter {
         return timelineArgs;
     }
 
+    protected toTestMessageArgs(...args: any[]): any[] {
+        let testItem: TestItem | undefined;
+        let testMessage: TestMessage | undefined;
+        for (const arg of args) {
+            if (TestItem.is(arg)) {
+                testItem = arg;
+            } else if (Array.isArray(arg)) {
+                testMessage = arg[0];
+            }
+        }
+        if (testMessage) {
+            const testItemReference = (testItem && testItem.controller) ? TestItemReference.create(testItem.controller.id, testItem.path) : undefined;
+            const testMessageDTO = {
+                message: testMessage.message,
+                actual: testMessage.actual,
+                expected: testMessage.expected,
+                contextValue: testMessage.contextValue,
+                location: testMessage.location ? fromLocation(testMessage.location) : undefined
+            };
+            return [TestMessageArg.create(testItemReference, testMessageDTO)];
```

Comment:
We can just return `[testItemReference, testMessageDTO]` and get rid of the TestMessageArgs namespace entirely. Just keep the code to handle TestMessageDTO

Reply:
We have to create a specific object for the commands on test message according to theia.d.ts:
The command will be called with an object containing:
         * - `test`: the {@link TestItem} the message is associated with, *if* it
         *    is still present in the {@link TestController.items} collection.
         * - `message`: the {@link TestMessage} instance.
So we still have to modify the argument processor on TestExtImpl to identify the 2 elements testItem and testMessage and create again this object. I thought it was easier to implement like it is currently. In the case we only pass the array  [testItemReference, testMessageDTO], we can't just map the elements of the array with toTestItem or toTestMessage, but we also need to return this new object. Or I may be missing an other processor that could do this on its own when invoking these specific commands.

Reply:
You are right, I did not notice that in the doc. How weird 🤷

Reply:
Yes, I totally agree with the weird arguments :)